presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java

```diff
@@ -95,9 +98,9 @@ public ConnectorSplitManager getSplitManager()
             }
 
             @Override
-            public ConnectorRecordSetProvider getRecordSetProvider()
+            public ConnectorPageSourceProvider getPageSourceProvider()
             {
-                return new TpchRecordSetProvider();
+                return new TpchPageSourceProvider(getMaxRowsPerPage(properties));
```

Comment:
I'm thinking we might want to make this a configuration option.  In the old code it tested the connector record set code that has a special test for `RecordPageSource` in Presto main that trigger different behavior.

Reply:
I will make record set the default behaviour